src/Instrumentation/PDO/src/PDOAttributeTracker.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\PDO;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+
+final class PDOAttributeTracker
+{
+    /**
+     * @var \WeakMap<\PDO, iterable<non-empty-string, bool|int|float|string|array|null>>
+     */
+    private \WeakMap $pdoToAttributesMap;
+    /**
+     * @var \WeakMap<\PDOStatement, int>
+     */
+    private \WeakMap $statementMapToPdoMap;
+
+    public function __construct()
+    {
+        /** @psalm-suppress PropertyTypeCoercion */
+        $this->pdoToAttributesMap = new \WeakMap();
+        /** @psalm-suppress PropertyTypeCoercion */
+        $this->statementMapToPdoMap = new \WeakMap();
+    }
+
+    public function trackStatementToPdoMapping(\PDOStatement $statement, \PDO $pdo)
+    {
+        $this->statementMapToPdoMap[$statement] = spl_object_id($pdo);
```

Comment:
Instead of storing the spl_object_id, couldn't you store a WeakRefeference to PDO directly and therefore making the lookup in trackedAttributesForStatement more trivial?

```php
$this->statementMapToPdoMap[$statement] = WeakReference::create($pdo);
```


Could look like this:

```php
$pdo = ($this->statementMapToPdoMap[$obj] ?? null)?->get();

return $this->pdoToAttributesMap[$pdo] ?? [];
```